docs/pages/badge.js

```diff
@@ -25,51 +24,255 @@ export default function BadgePage({ generatedDocGen }: {| generatedDocGen: DocGe
+                  </Table.Cell>
+                  <Table.Cell>
+                    <Badge text="Cancelled" type="error"/>
+                  </Table.Cell>
+                </Table.Row>
+                <Table.Row>
+                  <Table.Cell>
+                    <Text>Tattoo inspo</Text>
+                  </Table.Cell>
+                  <Table.Cell>
+                    <Badge text="Archived" type="neutral"/>
+                  </Table.Cell>
+                </Table.Row>
+              </Table.Body>
+            </Table>
+            `}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Use complex or verbose language. Instead use a single, scannable word. For example, New."
+            defaultCode={`
+            <Text size="300">7 wardrobe trends <Badge text="Fresh off the press"/></Text>
 
-      <Example
-        description="
-    The `Badge` component is rendered inline within parent element."
-        name="Example"
-        defaultCode={`
-<Text>Some text that uses Badge component as its child <Badge text="New" /></Text>
 `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection
+        name="Accessibility"
+        description={`
+        The badge text is read out by assistive technologies like screen readers so all users can access the meaning of the badge in context. Text should be clear enough to be understood immediately without relying on color alone.
+        `}
+      />
+      <MainSection
+        name="Localization"
+        description={`Be sure to localize the \`text\` prop. Note that localization can lengthen text by 20 to 30 percent. `}
       />
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Type"
+          description={`
+          Badge is available in five styles. Each \`type\` represents a messaging sentiment.
 
-      <Example
-        description="
-    Larger text example rendered with a top positioned `Badge`."
-        name="Example: positioning"
-        defaultCode={`
-  <Text size="600">Ads & Campaigns <Badge text="Beta" position="top"/></Text>
+          1. **Info (default)**
+          Communicates helpful information or an important attribute. For example, new, help.
+
+          2. **Success**
+          Indicates a constructive or successful state. For example, available, completed, approved, added.
+
+          3. **Warning**
+          Communicates cautionary or time-sensitive information to the user. For example, busy, missing, warning.
+
+          4. **Error**
+          Informs the user of problems or errors that require potential action to correct. For example, deleted, cancelled.
+
+          5. **Neutral**
+          Indicates a general, non-critical status update. For example, unavailable or not started.
+ `}
+        >
+          <MainSection.Card
+            defaultCode={`
+  <Table accessibilityLabel="Type examples">
```

Comment:
ditto here on using a function component to DRY out the markup

Reply:
I think leaving these examples simpler makes it easier for people to digest the code example, where we focus on the content rather than creating functions for these mini examples. I think it's easier to copy, paste and use the code when you don't need to worry about removing functions and state variables

Reply:
It also makes it easier for our designers to make updates to the docs